update auto architecture document in GitHub for 2023.0 release (#17141)
openvinotoolkit/openvino: src/plugins/auto/docs/architecture.md

* Select the highest-priority device capable of supporting the given model for LATENCY hint and THROUGHPUT hint. Or Select all devices capable of supporting the given model for CUMULATIVE THROUGHPUT hint.
* If model’s precision is FP32 but there is no device capable of supporting it, offload the model to a device supporting FP16.

The AUTO plugin is also default plugin for OpenVINO when user has not selected any device in the application.

## Specific Features of AUTO plugin

### Accelerating First Inference Latency (FIL)

Compiling the model to accelerator-optimized kernels may take some time. When AUTO select one accelerator, AUTO can start inference with the CPU of the system by default, as it provides very low latency and can start inference with no additional delays. While the CPU is performing inference, AUTO continues to load the model to the device best suited for the purpose and transfers the task to it when ready. 

Below is the example of CPU acceleration during GPU compilation. 
![alt text](https://docs.openvino.ai/latest/_images/autoplugin_accelerate.svg "AUTO cuts first inference latency (FIL) by running inference on the CPU until the GPU is ready")

> User can disable this acceleration feature by excluding CPU from the priority list or disable ov::intel_auto::enable_startup_fallback.The default value of ov::intel_auto::enable_startup_fallback is true.

Reviewer: ```suggestion
The user can disable this acceleration feature by excluding CPU from the priority list or disabling `ov::intel_auto::enable_startup_fallback`. Its default value is `true`.
```

### Inference on Multiple Devices

> The ov::hint::performance_mode property enables you to specify a performance option for AUTO to be more efficient for particular use cases. And the CUMULATIVE_THROUGHPUT hint enables running inference on multiple devices for higher throughput. 

Reviewer: ```suggestion
The `ov::hint::performance_mode` property enables you to specify a performance option for AUTO to be more efficient for particular use cases, while the `CUMULATIVE_THROUGHPUT` hint enables running inference on multiple devices at once for higher throughput.
```

> With CUMULATIVE_THROUGHPUT, AUTO loads the network model to all available devices in the candidate list, and then runs inference on them based on the devices priority.

Reviewer: ```suggestion
With CUMULATIVE_THROUGHPUT, AUTO loads the model to all available devices in the candidate list. Then, it runs inference, choosing devices based on their priority.
```

This is more than just rewording, see if it makes sense to make this change or not.
